spec/ameba/rule/lint/unused_pseudo_method_spec.cr

```diff
@@ -0,0 +1,175 @@
+require "../../../spec_helper"
+
+module Ameba::Rule::Lint
+  describe UnusedPseudoMethod do
+    subject = UnusedPseudoMethod.new
+
+    it "passes if typeof is unused" do
+      expect_no_issues subject, <<-CRYSTAL
+        typeof(1)
+        CRYSTAL
+    end
+
+    it "passes if as is unused" do
+      expect_no_issues subject, <<-CRYSTAL
+        as(Int32)
+        CRYSTAL
+    end
+
+    it "fails if pointerof is unused" do
+      expect_issue subject, <<-CRYSTAL
+        pointerof(Int32)
+        # ^^^^^^^^^^^^^^ error: Pseudo-method is not used
+        CRYSTAL
+    end
+
+    it "fails if sizeof is unused" do
+      expect_issue subject, <<-CRYSTAL
+        sizeof(Int32)
+        # ^^^^^^^^^^^ error: Pseudo-method is not used
+        CRYSTAL
+    end
+
+    it "fails if instance_sizeof is unused" do
+      expect_issue subject, <<-CRYSTAL
+        instance_sizeof(Int32)
+        # ^^^^^^^^^^^^^^^^^^^^ error: Pseudo-method is not used
+        CRYSTAL
+    end
+
+    it "fails if alignof is unused" do
+      expect_issue subject, <<-CRYSTAL
+        alignof(Int32)
+        # ^^^^^^^^^^^^ error: Pseudo-method is not used
+        CRYSTAL
+    end
+
+    it "fails if instance_alignof is unused" do
+      expect_issue subject, <<-CRYSTAL
+        instance_alignof(Int32)
+        # ^^^^^^^^^^^^^^^^^^^^^ error: Pseudo-method is not used
+        CRYSTAL
+    end
+
+    it "fails if offsetof is unused" do
+      expect_issue subject, <<-CRYSTAL
+        offsetof(Int32, 1)
+        # ^^^^^^^^^^^^^^^^ error: Pseudo-method is not used
+        CRYSTAL
+    end
+
+    it "fails if is_a? is unused" do
+      expect_issue subject, <<-CRYSTAL
+        is_a?(Int32)
+        # ^^^^^^^^^^ error: Pseudo-method is not used
+        CRYSTAL
+    end
+
+    it "fails if as? is unused" do
+      expect_issue subject, <<-CRYSTAL
+        as?(Int32)
+        # ^^^^^^^^ error: Pseudo-method is not used
+        CRYSTAL
+    end
+
+    it "fails if responds_to? is unused" do
+      expect_issue subject, <<-CRYSTAL
+        responds_to?(:foo)
+        # ^^^^^^^^^^^^^^^^ error: Pseudo-method is not used
+        CRYSTAL
+    end
+
+    it "fails if nil? is unused" do
+      expect_issue subject, <<-CRYSTAL
+        nil?
+        # ^^ error: Pseudo-method is not used
+        CRYSTAL
+    end
+
+    it "fails if prefix not is unused" do
+      expect_issue subject, <<-CRYSTAL
+        !true
+        # ^^^ error: Pseudo-method is not used
+        CRYSTAL
+    end
```

Comment:
isn't this confusing? It says "Pseudo-method is not used" but there are not any methods/calls.

Reply:
`!` is a pseudo method call

Reply:
`Not` should be IMO handled differently so it will report just the `!` and not the whole node.

Reply:
"Return value of pseudo-method call is unused"?

Reply:
Or `Pseudo-method call to `%s` is not used`?